src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderHistory/Services/OrderService.ts

```diff
@@ -27,8 +30,24 @@ module Orckestra.Composer {
                 });
         }
 
-        public saveEditOrder() {
-            return this.orderRepository.saveEditOrder();
+        public saveEditOrder(orderNumber: string) {
+            return this.orderRepository.saveEditOrder(orderNumber).then((result: ICompleteCheckoutResult) => {
+
+                result.IsUpdatedOrder = true;
+                this.cacheProvider.defaultCache.set(this.orderCacheKey, result).done();
+                this.cacheProvider.defaultCache.set(this.orderConfirmationCacheKey, result).done();
+
+                this.eventHub.publish(MyAccountEvents.EditOrderFinished, { data: { orderNumber } });
+
+                this.cartService.getFreshCart(true);
+
+                if (result.NextStepUrl) {
+                    window.location.href = result.NextStepUrl;
+                }
+
+            })
+                .then(cart => this.eventHub.publish(CartEvents.CartUpdated, { data: cart }))
```

Comment:
Is this executed? I think no, as you do not return cart
you need to do 
return this.cartService.getFreshCart(true);

also do this.cartService.invalidateCache;